plugins/module_utils/job.py

```diff
@@ -76,9 +73,52 @@ def _get_job_output(job_id="*", owner="*", job_name="*", dd_name=""):
                 str(rc), str(err)
             )
         )
-    if not out:
-        raise RuntimeError("Failed to retrieve job output. No job output found.")
-    jobs = _parse_jobs(out)
+    jobs = []
+    if out:
+        jobs = _parse_jobs(out)
+    if not jobs:
+        jobs = _job_not_found(job_id, owner, job_name, dd_name)
+
+    return jobs
+
+
+def _job_not_found(job_id, owner, job_name, dd_name, ovrr=None):
+    jobs = []
+
+    job = {}
+
+    job["job_id"] = job_id
+    job["job_name"] = job_name
+    job["subsystem"] = None
+    job["system"] = None
+    job["owner"] = None
+
+    job["ret_code"] = {}
+    job["ret_code"]["msg"] = "Job Not Found"
+    job["ret_code"]["code"] = None
+    job["ret_code"]["msg_code"] = "00"
```

Comment:
Lets chat about this, because according to our contract/doc , we should return None for msg_code , see below "if and only if we can convert msg_code into an integer value we return it otherwise we return None" 

- msg_code - would the string value of the job error reduced down to something more consumable such as "00" ("00" | "04" | "08" | "S0C4"

- code - if and only if we can convert msg_code into an integer value we return it otherwise we return None
Something like msg_code :"00" can be converted to type int: code : 00
Something like msg_code :"ABEND", msg_code :"ABEND(S1234)", msg_code :"CANCELED", can not convert thus return None

Reply:
I think we agree code should be None, and the question is should msg_code be a simple integer?  Would changing the msg_code to "NOT FOUND" make sense?  I don't recall a length limit on that field, and that would become a string, not an integer, similar to 'SOC4".